tests/unit/test_coordinate_utilities.py

```diff
@@ -0,0 +1,431 @@
+        ]
+
+        with self.subTest(
+            'Retrieves expected override dimensions for the science variable'
+        ):
+            self.assertEqual(
+                get_override_projected_dimensions(
+                    self.varinfo, '/Soil_Moisture_Retrieval_Data_AM/surface_flag'
+                ),
+                expected_override_dimensions_AM,
+            )
+
+        with self.subTest(
+            'Retrieves expected override dimensions for the longitude variable'
+        ):
+            self.assertEqual(
+                get_override_projected_dimensions(self.varinfo, self.longitude),
+                expected_override_dimensions_AM,
+            )
+
+        with self.subTest(
+            'Retrieves expected override dimensions for the latitude variable'
+        ):
+            self.assertEqual(
+                get_override_projected_dimensions(self.varinfo, self.latitude),
+                expected_override_dimensions_AM,
+            )
+
+        with self.subTest(
+            'Retrieves expected override dimensions science variable with a different grid'
+        ):
+            self.assertEqual(
+                get_override_projected_dimensions(
+                    self.varinfo, '/Soil_Moisture_Retrieval_Data_PM/surface_flag_pm'
+                ),
+                expected_override_dimensions_PM,
+            )
+
+    def test_get_row_col_sizes_from_coordinate_datasets(self):
+        """Ensure that the correct row and column sizes are
+        returned for the requested coordinates
+        """
+        expected_row_col_sizes = (5, 10)
+        lat_1d_array = np.array([1, 2, 3, 4])
+        lon_1d_array = np.array([6, 7, 8, 9])
+        lat_irregular_array = np.array([[1, 2, 3], [3, 4, 5]])
+        lon_irregular_array = np.array([[6, 7], [8, 9], [10, 11]])
+        with self.subTest('Retrieves the expected row col sizes from the coordinates'):
+            self.assertEqual(
+                get_row_col_sizes_from_coordinate_datasets(self.lat_arr, self.lon_arr),
+                expected_row_col_sizes,
+            )
+        with self.subTest('Retrieves the expected row col sizes for the 1d array'):
+            self.assertEqual(
+                get_row_col_sizes_from_coordinate_datasets(lat_1d_array, lon_1d_array),
+                (4, 4),
+            )
+
+        with self.subTest(
+            'Raises an exception when the lat and lon array shapes do not match'
+        ):
+            with self.assertRaises(IrregularCoordinateVariables) as context:
+                get_row_col_sizes_from_coordinate_datasets(
+                    lat_irregular_array, lon_irregular_array
+                )
+            self.assertEqual(
+                context.exception.message,
+                f'Longitude coordinate shape: "{lon_irregular_array.shape}"'
+                f'does not match the latitude coordinate shape: "{lat_irregular_array.shape}"',
+            )
```

Comment:
I would use the `self.assertRaisesRegex` class method to check exception messages.